### PR TITLE
Mini AOD 74 X

### PR DESCRIPTION
**Pull request by larsonka**
Changes to allow the analysis to be run over miniAOD.

Simply change the flag in the Zprime2muAnalysis_cfg.py file and you can run the analysis and main histogramming modules as normal.

### CODE REVIEW

File: python/MuonPhotonMatch_cff_miniAOD.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# selected refit manages to recover, the photon match may be
+# wrong. The photon brem recovery needs to be studied more anyway.
+muonPhotonMatch = cms.EDProducer('TrivialDeltaRViewMatcher',
+                                 src     = cms.InputTag('slimmedMuons'),
+                                 matched = cms.InputTag('slimmedPhotons'),
+                                 distMin = cms.double(0.1)
+                                 )
+
+def addUserData(patMuonProducer, tag=cms.InputTag('muonPhotonMatch')):
+    patMuonProducer.userData.userCands.src.append(tag)
```
Comment (BenjaminRS) [inline python/MuonPhotonMatch_cff_miniAOD.py:16]: Is there a chance of confusion with the muonPhotonMatch in python/MuonPhotonMatch_cff.py of the same name?
why not use something like the following in this file:
import MuonPhotonMatch_cff.py
muonPhotonMatch_miniAOD = MuonPhotonMatch_cff.muonPhotonMatch.clone()
then change the src and matched values of muonPhotonMatch_miniAOD?

Comment (larsonka) [inline python/MuonPhotonMatch_cff_miniAOD.py:16]: There shouldn't be any confusion with it unless both muonPhotonMatch_cff_miniAOD.py and MuonPhotonMatch_cff.py are both imported for some reason, which does not happen in any of the modifications I've made. 

I think it does make sense from a logical standpoint though, and I don't think it will be difficult to change. I will try to implement it and see if there are any issues.

File: python/Zprime2muAnalysis_cfg.py [hunk 2]
```diff
@@ -21,4 +21,8 @@
 from Configuration.AlCa.GlobalTag_condDBv2 import GlobalTag
 process.GlobalTag.globaltag = 'MCRUN2_74_V9'
 
-process.load('SUSYBSMAnalysis.Zprime2muAnalysis.Zprime2muAnalysis_cff')
+flag = cms.string('miniAOD')
```
Comment (BenjaminRS) [inline python/Zprime2muAnalysis_cfg.py:24]: Better to leave the default as AOD

File: python/goodData_cff.py [hunk 3]
```diff
@@ -10,7 +10,7 @@
                                   L1GtReadoutRecordTag = cms.InputTag('gtDigis')
                                   )
 
-primaryVertexFilter = cms.EDFilter('GoodVertexFilter',
+primaryVertex = cms.EDFilter('GoodVertexFilter',
```
Comment (BenjaminRS) [inline python/goodData_cff.py]: Why the change of name?

Comment (rradogna) [inline python/goodData_cff.py]: Hi Guys,

sorry for the delay in answering, but as you know we have tight dead lines.
Indeedmy proposal is to wait a couple of weeks before taking decisions
about this PR, just to have enough time for deep checks.

b.t.w...
quickly looking at the code I don’t aspect big problems since there are new
files (for miniAOD) and it seams that Kurt is not changing to much in
common files.

possible issues:
1. Kurt, maybe wrongly, changed the name of a filter that I changed to fix
a problem I recently had compiling in 7_4_12:
https://github.com/cms-analysis/SUSYBSMAnalysis
-Zprime2muAnalysis/pull/14/files#diff-eeac8088a38b3a15a6cd08c0080bf702L13
(question from benjmin in previous mail)

2.I want to figure out how these flags work and if this is the only file
where you can switch between miniAOD or AOD:
https://github.com/larsonka/SUSYBSMAnalysis-Zprime2muAnalysis/blob/MiniAOD
_74X/python/Zprime2muAnalysis_cfg.py
for sure, as suggested by Benjamin, it's better to have the AOD as default

Did you, Kurt, try to compile your code and produce a small pattuple? We
should check that it works. I don't think it will work because of the
possible issue 1. in CMSSW_4_12. But I could be wrong.

We should try to manage these checks before merging the PR.

ciao,
Raffaella

2015-11-17 23:47 GMT+01:00 Benjamin Radburn-Smith notifications@github.com
:

Comment (alfloren) [inline python/goodData_cff.py]: HI,

I agree with Raffaella, let’s not spend time on that right now. Better after the approval.

Alice

**Comment by larsonka** [comment]
Yes, the approval is more important for sure.

I've made changes as Benjamin asked, and I will set the default to AOD initially.

The primary vertex filter issue was preventing me from compiling as you said. I will look into it more and come up with a clearer explanation of the problem and let you know. Right now, I am adding the new data to the AN and working on other analysis issues, so I won't have an answer until the weekend at least I think.

I previously was able to compile and make PAT tuples. This was before any changes I made recently, so I will test it again once I make all the changes suggested so far.

**Comment by larsonka** [comment]
So for the primary vertex filter issue, I changed the name of the module because it was conflicting with the METFilter path run during the PAT tupling process (as can be seen in the error message below).

To fix this, I changed the module name in goodData_cff.py and also changed in in python/PATTuple_cfg.py where it is used. These are the only places it is referenced as far as I can tell.

With these changes, I am able to run the PAT tupling code as normal and produce histos, graphs, etc.

(This was all done in CMSSW_7_4_12).

----- Begin Fatal Exception 30-Nov-2015 20:00:45 CET-----------------------

An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named tuple_mc.py
Exception Message:
python encountered the error: <type 'exceptions.ValueError'>
Trying to override definition of primaryVertexFilter while it is used by the sequence metFilters
 new object defined in: /afs/cern.ch/user/k/klarson/public/working/official/CMSSW_7_4_12/python/SUSYBSMAnalysis/Zprime2muAnalysis/goodData_cff.py
 existing object defined in: /cvmfs/cms.cern.ch/slc6_amd64_gcc491/cms/cmssw/CMSSW_7_4_12/python/RecoMET/METFilters/primaryVertexFilter_cfi.py
----- End Fatal Exception -------------------------------------------------
